File: tasks/javascript/6kyu/CountCharactersInYourString.js
// Count characters in your string

// The main idea is to count all the occurring characters in a string. If you have a string like aba, then the result should be {'a': 2, 'b': 1}.

// What if the string is empty? Then the result should be empty object literal, {}.

// Sample Tests:
// const chai = require('chai');
// const assert = chai.assert;
// chai.config.truncateThreshold = 0;

// describe('Sample Tests', function() {
//   function test(string, expected) {
//     it(`<code>'${string}'</code>`, function() {
//       assert.deepEqual(count(string), expected);
//     });
//   }

//   test('', {});
//   test('a', {'a': 1});
//   test('ab', {'a': 1, 'b': 1});
//   test('aba', {'a': 2, 'b': 1});
//   test('ABC', {'A': 1, 'B': 1, 'C': 1});
// });

function count(string) {
  let letters = {};
  string = string.split("");
  for (let i = 0; i < string.length; i++) {
    if (!letters.hasOwnProperty(string[i])) letters[`${string[i]}`] = 1;
    else letters[`${string[i]}`]++;
  }
  return letters;
}

File: tasks/java/6kyu/CountCharactersInYourString.java
// Count characters in your string

// The main idea is to count all the occurring characters in a string. If you have a string like aba, then the result should be {'a': 2, 'b': 1}.

// What if the string is empty? Then the result should be empty object literal, {}.

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class CountCharactersInYourString {
    public static Map<Character, Integer> count(String str) {
        Map<Character, Integer> mapRes = new HashMap<Character, Integer >();
        char[] charArr = str.toCharArray();
        for (int i = 0; i < charArr.length; i++) {
            int counter = 0;
            char first = charArr[i];
            for (int j = 0; j < charArr.length; j++) {
                if(first == charArr[j]) {
                    counter++;
                }
            }      
            mapRes.put(charArr[i], counter);
        }
        return mapRes;
    }
    public static void main(String[] args) {
        
    }

}
